src/pages/patientView/mutation/PatientViewCnaDataStore.ts

```diff
@@ -0,0 +1,81 @@
+import { SimpleGetterLazyMobXTableApplicationDataStore } from '../../../shared/lib/ILazyMobXTableApplicationDataStore';
+import { DiscreteCopyNumberData } from 'cbioportal-ts-api-client';
+import { action, computed, observable, makeObservable } from 'mobx';
+import _ from 'lodash';
+import PatientViewUrlWrapper from '../PatientViewUrlWrapper';
+
+function cnaMatch(d: DiscreteCopyNumberData[], id: DiscreteCopyNumberData) {
+    return (
+        d[0].entrezGeneId === id.entrezGeneId &&
+        d[0].gene.hugoGeneSymbol === id.gene.hugoGeneSymbol
+    );
+}
+
+function cnaIdKey(c: DiscreteCopyNumberData) {
+    return `{ "entrezGeneId": "${c.entrezGeneId}", "hugoGeneSymbol": "${c.gene.hugoGeneSymbol}" }`;
+}
+
+type CnaIdKey = string;
+
+export default class PatientViewCnaDataStore extends SimpleGetterLazyMobXTableApplicationDataStore<
+    DiscreteCopyNumberData[]
+> {
+    @observable mouseOverCna: Readonly<DiscreteCopyNumberData> | null = null;
+    private selectedCnasMap = observable.map<string, DiscreteCopyNumberData>();
+
+    public get onlyShowSelectedInTable() {
+        return (
+            this.urlWrapper.query.genomicEvolutionSettings
+                .showOnlySelectedCnasInTable === 'true'
+        );
+    }
+
+    @action
+    public toggleSelectedCna(c: Readonly<DiscreteCopyNumberData>) {
+        const key = cnaIdKey(c);
+        if (this.selectedCnasMap.has(key)) {
+            this.selectedCnasMap.delete(key);
+        } else {
+            this.selectedCnasMap.set(key, c);
+        }
+    }
+
+    @action
+    public setSelectedCna(cna: Readonly<DiscreteCopyNumberData[]>) {
+        this.selectedCnasMap.clear();
+        let count = 0;
+        for (const c of cna) {
+            this.toggleSelectedCna(c);
+            count += 1;
+        }
+    }
+
+    @computed public get selectedCna(): Readonly<DiscreteCopyNumberData[]> {
+        return Array.from(this.selectedCnasMap.values());
+    }
+
+    public isCnaSelected(c: DiscreteCopyNumberData) {
+        return this.selectedCnasMap.has(cnaIdKey(c));
+    }
+
+    constructor(
+        getData: () => DiscreteCopyNumberData[][],
+        private urlWrapper: PatientViewUrlWrapper
+    ) {
+        super(getData);
+
+        makeObservable(this);
+
+        this.dataHighlighter = (mergedCna: DiscreteCopyNumberData[]) => {
+            const highlightedCnas = [];
+            if (!this.onlyShowSelectedInTable) {
+                // dont put highlight on selected cnas if those are all we're showing
+                highlightedCnas.push(...this.selectedCna);
+            }
+            if (this.mouseOverCna) {
+                highlightedCnas.push(this.mouseOverCna);
+            }
```

Comment:
might be better to simply do this:

```suggestion
          highlightedCnas.push(...this.selectedCna)
```